pkg/reconciler/v1alpha1/service/service.go

```diff
@@ -213,20 +214,64 @@ func (c *Reconciler) reconcile(ctx context.Context, service *v1alpha1.Service) e
 	// Apply additional logic, once the generic data has been propagated.
 	switch {
 	case service.Spec.RunLatest != nil:
-		runLatestCheckTrafficMigrated(ss, &route.Status, &config.Status)
+		c.runLatestCheckTrafficMigrated(service, &route.Status, &config.Status)
+	case service.Spec.Release != nil:
+		c.releaseCheckTrafficMigrated(service, &route.Status, &config.Status)
```

Comment:
Couldn't we replace this whole `switch` (and both functions) with:
```go
if rc := service.Status.GetCondition(v1alpha1.ServiceConditionRoutesReady); rc != nil && rc.Status == corev1.ConditionTrue {
   want := route.Spec.Traffic.DeepCopy()
   for idx := range want {
      if want[idx].ConfigurationName == config.Name {
         want[idx].RevisionName = config.LatestReadyConfigurationName
         want[idx].ConfigurationName = ""
      }
   }
   if got := route.Status.Traffic; !cmp.Equal(got, want) {
      s.Status.MarkRouteNotYetReady()
   }
}
```

Reply:
As is this definitely makes old tests fail. New tests still pass.

Reply:
FWIW `cmp.Equal` can panic. If you go with this approach, consider: https://github.com/knative/pkg/blob/2a1c043ce3d778a4017e9c988b7a7dd0644312ad/kmp/diff.go#L53